drivers/sensor/bh1749/bh1749.c

```diff
@@ -217,58 +213,59 @@ static void bh1749_async_init(struct k_work *work)
 			data->ready = true;
 			LOG_INF("BH1749 initialized");
 		} else {
-			k_work_schedule(&bh1749_init_work,
+			k_work_schedule(init_work,
 					K_MSEC(async_init_delay[data->async_init_step]));
 		}
 	}
 }
 
-static int bh1749_async_init_reset_check(struct bh1749_data *data)
+static int bh1749_async_init_reset_check(const struct device *dev)
 {
-	(void)memset(data->sample_rgb_ir, 0, sizeof(data->sample_rgb_ir));
+	struct bh1749_data *data = dev->data;
+	const struct bh1749_config *config = dev->config;
+	const struct i2c_dt_spec *i2c = &config->i2c;
 	int err;
 
-	err = bh1749_sw_reset(data->i2c);
+	(void)memset(data->sample_rgb_ir, 0, sizeof(data->sample_rgb_ir));
 
-	if (err) {
-		LOG_ERR("Could not apply software reset.");
-		return -EIO;
+	err = bh1749_sw_reset(i2c);
+	if (err < 0) {
+		LOG_ERR("Could not apply software reset");
+		return err;
 	}
-	err = bh1749_check(data);
-	if (err) {
-		LOG_ERR("Communication with BH1749 failed with error %d", err);
-		return -EIO;
-	}
-	return 0;
+
+	return bh1749_check(i2c);
 }
 
-static int bh1749_async_init_rgb_enable(struct bh1749_data *data)
+static int bh1749_async_init_rgb_enable(const struct device *dev)
 {
+	const struct bh1749_config *config = dev->config;
 	int err;
 
-	err = bh1749_rgb_measurement_enable(data, true);
-	if (err) {
+	err = bh1749_rgb_measurement_enable(&config->i2c, true);
```

Comment:
nit: for consistency either use `&config->i2c` or `i2c` throughout the driver

Reply:
>  nit: for consistency either use &config->i2c or i2c throughout the driver

That's fine, but the rule I was following (consistently) was to store the pointer in a local if and only if I needed it more than once in the same function.